File: Games/NoughtsAndCrosses/src/main/java/jokatu/game/games/noughtsandcrosses/event/CellChosenEventHandler.java
package jokatu.game.games.noughtsandcrosses.event;

import jokatu.game.Game;
import jokatu.game.event.SpecificEventHandler;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import static java.text.MessageFormat.format;

/**
 * Private events should be forwarded to the users they specify.
 * @author Steven Weston
 */
@Component
public class CellChosenEventHandler extends SpecificEventHandler<CellChosenEvent> {

	@NotNull
	@Override
	protected Class<CellChosenEvent> getEventClass() {
		return CellChosenEvent.class;
	}

	@Override
	protected void handleCastEvent(@NotNull Game<?> game, @NotNull CellChosenEvent event) {

		sender.send(
				format("/topic/substatus.game.{0}.cell_{1}", game.getIdentifier(), event.getCell()),
				event.getNoughtOrCross()
		);
		sender.send(
				format("/topic/substatus.game.{0}.lines", game.getIdentifier()),
				event.getLines()
		);
	}
}

File: Games/NoughtsAndCrosses/src/main/kotlin/jokatu/game/games/noughtsandcrosses/event/CellChosenEventHandler.kt
package jokatu.game.games.noughtsandcrosses.event

import jokatu.components.stomp.Topic
import jokatu.game.Game
import jokatu.game.event.SpecificEventHandler
import jokatu.game.games.noughtsandcrosses.game.Line
import org.springframework.stereotype.Component

/**
 * Private events should be forwarded to the users they specify.
 * @author Steven Weston
 */
@Component
class CellChosenEventHandler : SpecificEventHandler<CellChosenEvent>(CellChosenEvent::class) {
	override fun handleCastEvent(game: Game<*>, event: CellChosenEvent) {
		sender.send(Cell(game, event.cell), event.getNoughtOrCross())
		sender.send(Lines(game), event.lines)
	}
}

private class Cell(game: Game<*>, cell: Int): Topic<String>("substatus.game.${game.identifier}.cell_$cell")

private class Lines(game: Game<*>): Topic<List<Line>>("substatus.game.${game.identifier}.lines")